include/views/view.h

```diff
@@ -130,79 +146,113 @@ struct MatrixView {
   using container_t = view_container_t;
   using index_t = view_index_t;
   using self_t = MatrixView<value_t, container_t, index_t, layout>;
-  container_t &data_;
-  index_t size_data_;  // real size of the data
+  container_t data_;
   index_t sizeR_;      // number of rows
   index_t sizeC_;      // number of columns
   index_t sizeL_;      // size of the leading dimension
-  index_t disp_;       // displacementt od the first element
+
+  // global pointer access inside the kernel
+  cl::sycl::global_ptr<value_t> ptr_;
+
   // UPLO, BAND(KU,KL), PACKED, SIDE ARE ONLY REQUIRED
-  MatrixView(view_container_t &data, view_index_t sizeR, view_index_t sizeC);
-  MatrixView(view_container_t &data, view_index_t sizeR, view_index_t sizeC,
-             view_index_t sizeL, view_index_t disp);
+  MatrixView(view_container_t data, view_index_t sizeR, view_index_t sizeC);
+  MatrixView(view_container_t data, view_index_t sizeR, view_index_t sizeC,
+             view_index_t sizeL);
   MatrixView(
       MatrixView<view_value_t, view_container_t, view_index_t, layout> opM,
-      view_index_t sizeR, view_index_t sizeC, view_index_t sizeL,
-      view_index_t disp);
+      view_index_t sizeR, view_index_t sizeC, view_index_t sizeL);
 
   /*!
    * @brief Returns the container
    */
-  container_t &get_data();
+  SYCL_BLAS_INLINE container_t get_data();
 
   /*!
    * @brief Returns the container
    */
-  value_t *get_pointer();
+  SYCL_BLAS_INLINE value_t *get_pointer();
 
   /*!
-   * @brief Returns the data size
+   * @brief Returns the size of the view.
    */
-  index_t get_data_size() const;
+  SYCL_BLAS_INLINE index_t get_size() const;
 
   /*!
-   * @brief Returns the size of the view.
+   * @brief Returns the leading dimension.
    */
-  inline index_t get_size() const;
+  SYCL_BLAS_INLINE const index_t getSizeL() const;
 
   /*! get_size_row.
    * @brief Return the number of columns.
    * @bug This value should change depending on the access mode, but
    * is currently set to Rows.
    */
-  inline index_t get_size_row() const;
+  SYCL_BLAS_INLINE index_t get_size_row() const;
 
   /*! get_size_col.
    * @brief Return the number of columns.
    * @bug This value should change depending on the access mode, but
    * is currently set to Rows.
    */
-  index_t get_size_col() const;
-
-  /*! get_access_device.
-   * @brief Access on the Device (e.g CPU: Row, GPU: Column).
-   */
-  int get_access_device() const;
+  SYCL_BLAS_INLINE index_t get_size_col() const;
 
   /*! adjust_access_displacement.
    * @brief set displacement from the origin.
+   * For USM case, this method is not going to do anything as the library
+   * doesn't allow pointer manipulation.
```

Comment:
I suggest integrating the comment as follows for making things clearer.
```suggestion
   * This method allows to have a pointer arithmetic semantics for buffers
   * in the host code. The end result of the pointer arithmetic is passed  
   * as an access displacement for the buffer.
   * In the USM case, this method is not going to do anything since the
   * pointer arithmetic is performed implicitly.
```

Reply:
This has been addressed in https://github.com/codeplaysoftware/sycl-blas/pull/441